### PR TITLE
Replace gp_url_login/logout() with wp_login/logout_url()

### LINKED ISSUE #75: Trying to log out shows Are you sure? message
GP_Route_Login redirects to a URL that `wp_logout_url()` returns. That URL is HTML encoded, so query parameters, especially `_wpnonce`, will not get read right.

Nonces won't help, if they never get printed to a web page, so instead of using the logout route, the logout link should point directly to wp-login.php using the wp_logout_url() function. Or, if the route is deemed to be worth saving, the logout link should have a nonce in it and the final WordPress logout url should be manually constructed.

### PR DESCRIPTION
**Pull request by toolstack**
Includes removing the login/logout routes.

Resolves #75.

### CODE REVIEW

**Comment by bjork** [comment]
This works fine in my testing. One note: `wp_login_url()` does not seem to do HTML encoding. It does not currently technically need it, since it only returns a link with one URL parameter, but I think we should encode it nevertheless.

(I created a ticket on the missing documentation of the encoding: https://core.trac.wordpress.org/ticket/34352)

File: gp-templates/helper-functions.php [hunk 1]
```diff
@@ -86,7 +86,7 @@ function textareas( $entry, $permissions, $index = 0 ) {
 				_e('You are not allowed to edit this translation.');
 			}
 			else {
-				printf( __('You <a href="%s">have to log in</a> to edit this translation.'), gp_url_login() );
+				printf( __('You <a href="%s">have to log in</a> to edit this translation.'), wp_login_url() );
```
Comment (ocean90) [inline gp-templates/helper-functions.php]: Should be `esc_url( wp_login_url() )`.

File: gp-templates/translation-row.php [hunk 2]
```diff
@@ -36,7 +36,7 @@
 			$edit_text = __('You are not allowed to add a translation.');
 		}
 		else {
-			$edit_text = sprintf( __('You <a href="%s">have to login</a> to add a translation.'), gp_url_login() );
+			$edit_text = sprintf( __('You <a href="%s">have to login</a> to add a translation.'), wp_login_url() );
```
Comment (ocean90) [inline gp-templates/translation-row.php]: Should be `esc_url( wp_login_url() )`.

File: gp-includes/template.php [hunk 3]
```diff
@@ -73,10 +73,10 @@ function gp_nav_menu_items( $location = 'main' ) {
 			$user = GP::$user->current();
 
 			$items[ gp_url( '/profile' ) ] = __('Profile');
-			$items[ gp_url( '/logout' ) ]  = __('Log out');
+			$items[ wp_logout_url( gp_url_current() ) ]  = __('Log out');
 		}
 		else {
-			$items[ gp_url_login() ] = __('Log in');
+			$items[ wp_login_url( gp_url_current() ) ] = __('Log in');
```
Comment (ocean90) [inline gp-includes/template.php]: Looks like none of the links are escaped currently. Maybe we should do this in `gp_nav_menu()`. (Something for a new issue/PR.)

**Comment by toolstack** [comment]
@ocean90 I just followed the WP documentation on how to use wp_login/logout_url() :wink:.

Neither use esc_url() in the examples, however as @bjork did open a ticket about that so I'm not sure which way we want to go, fix our code or fix core.

**Comment by ocean90** [comment]
@toolstack Which documentation?

**Comment by toolstack** [comment]
The WP Codex:  http://codex.wordpress.org/Function_Reference/wp_login_url

**Comment by bjork** [comment]
I don't think anything else will be fixed in core than the doc block documentation, because that would break a lot of implementations, if suddenly `wp_login_url()` would start returning an encoded url. So I would just fix it in GlotPress.

**Comment by toolstack** [comment]
@bjork I agree that's the "path of least resistance" but I suspect this is an issue in a lot of other code as well so the "right" thing might be to fix it in core.  Afterall the URL should never NOT be encoded so why not do it in the function.

**Comment by ocean90** [comment]
@toolstack Well, as you see, the codex isn't always right. (Why does it still exists…)

Since there is currently no plan to _fix_ this in core we have to escape it by ourself. Late-escaping.

**Comment by toolstack** [comment]
I'm fine with that, just wanted to make sure we're all in agreement on it.

I'll update the PR.

**Comment by bjork** [comment]
@toolstack Good point. As the function is one of the so called template functions, most people probably have assumed it is behaving differently than it actually is. A change like that _could_ be possible. 

There are a lot of reasons for an URL to be not encoded. Even in the presentation layer. For HTML output though, we need encoding.

**Comment by bjork** [comment]
Only wp_login_url needs to be escaped, wp_logout_url does include it.

**Comment by toolstack** [comment]
As per the discussion on slack (https://wordpress.slack.com/archives/glotpress/p1445521678000331) we'll leave the esc_url() on both login and logout.

**Pull request merged**